File: Easy/TwoSum.js
"Link: https://leetcode.com/problems/two-sum/"
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let table = {};
    for (let i in nums) {
        table[nums[i]] = i;
    }
    for (let i in nums) {
        let check = table[target - nums[i]];
        if (check && check != i) return [i, check];
    }
};

File: Two-Sum/Two-Sum.cpp
class Solution {
public:
    vector&lt;int&gt; twoSum(vector&lt;int&gt;&amp; nums, int target) {
        vector&lt;int&gt; result;
        for (int i = 0; i &lt; nums.size(); i++) {
            for (int j = 0; j &lt; nums.size(); j++) {
                if (nums[i] + nums[j] == target &amp;&amp; i != j) {
                    result.push_back(i);
                    result.push_back(j);
                    return result;
                }
            }
        }
        return nums;
    }
};